flaskApi/parser.py

```diff
@@ -0,0 +1,64 @@
+from pptx import Presentation
+
+
+def get_presentation_as_list_of_slides(path_to_presentation):
+    """
+    Gets   pptx presentation and read it and return it as list of slides
+
+   Parameters
+   ----------
+   path_to_presentation : str
+        path to pptx presentation
+
+   Returns
+   -------
+   list
+        return the presentation as list of dicts
+        example:
+        [
+           mocks\n
+           Python mocks are objects that ' ,
+
+         ' Refactoring code\n
+           The process of improving the design ...' ,
+           ...
+       ]
+   """
+    presentation = Presentation(path_to_presentation)
+
+    presentation_as_list_of_slides = [get_text_from_slide(slide) for slide in presentation.slides]
+    return presentation_as_list_of_slides
+
+
+def get_text_from_slide(slide):
+    """
+       Gets  slide and return it as string
+
+      Parameters
+      ----------
+       slide :  pptx.slide.Slide object
+
+      Returns
+      -------
+      string
+           return the slide as string
+            example:
+
+            '  Refactoring code\n
+              The process of improving the design ...'
+      """
+    slide_as_list_of_texts_box = []
+
+    for shape in slide.shapes:
+        if not shape.has_text_frame:
+            continue
+
+        for paragraph in shape.text_frame.paragraphs:
+
+          if not paragraph.text:
+                continue
+
+          s = f'{paragraph.text}'.strip()
```

Comment:
This name does not adhere to python's naming conventions. Names should be indicative, and when used properly can reduce redundant documentation. In addition, each Variable, function, module, constant and class names has a specific naming convention. Read this link to learn the proper conventions - https://visualgit.readthedocs.io/en/latest/pages/naming_convention.html. If you don't know what CamelCase and snake_case means, read this link - https://www.google.com/amp/s/www.theserverside.com/answer/Camel-case-vs-snake-case-Whats-the-difference%3famp=1 . For a general overview on python conventions (PEP-8), read this link - https://peps.python.org/pep-0008